share/availability/discovery/set.go

```diff
@@ -47,6 +47,7 @@ func (ps *limitedSet) Size() uint {
 func (ps *limitedSet) Add(p peer.ID) (added bool) {
 	ps.lk.Lock()
 	if _, ok := ps.ps[p]; ok {
+		ps.lk.Unlock()
```

Comment:
This

Reply:
Nice find

Reply:
Could be one unlock:
```
ps.lk.Lock()
_, ok := ps.ps[p]
ps.lk.Unlock()
if ok{...
```

Reply:
No, because we write to the map afterwards